agent/task_context.go

```diff
@@ -194,11 +229,12 @@ func (tc *taskContext) getExecTimeout() time.Duration {
 
 // commandBlock contains information for a block of commands.
 type commandBlock struct {
-	block       command.BlockType
-	commands    *model.YAMLCommandSet
-	timeoutKind timeoutType
-	getTimeout  func() time.Duration
-	canFailTask bool
+	block               command.BlockType
+	commands            *model.YAMLCommandSet
+	timeoutKind         timeoutType
+	getTimeout          func() time.Duration
+	canTimeoutHeartbeat bool
```

Comment:
canHeartbeatTimeout?  unless i'm confused about what this is

Reply:
I'm using time out as a verb instead of noun, but I can change it to `TimeOut` to make it clearer.